src/olmo_core/train/trainer.py

```diff
@@ -659,29 +659,29 @@ def fit(self):
 
         log.info(f"Training for {self.max_steps:,d} steps")
 
-        for callback in self._iter_callbacks():
-            callback.pre_train()
-        self.train_module.pre_train()
-
-        barrier()
-
-        # Quick check if the run has already been canceled.
-        if self.is_canceled:
-            self._shutdown()
-            return
-
         # Install SIGTERM + SIGINT handlers.
         og_sigterm_handler = signal.signal(signal.SIGTERM, self._handle_os_signal)
         og_sigint_handler = signal.signal(signal.SIGINT, self._handle_os_signal)
 
-        # Do a dry-run for compiling and catch OOMs.
-        self._dry_run_batch()
-
         try:
+            for callback in self._iter_callbacks():
```

Comment:
Moved this stuff down here so that `callback.on_error()` will be called if any of these steps fail.